Secure review - `v1.01` release

CharlVS opened the pull request:
Secure code review of the newly created KomodoPlatform `dragon_logs` flutter package hosted at  https://pub.dev/packages/dragon_logs. This removes a lot of logging-related code from the main project and handles it here much more efficiently using a novel approach for web storage.

This is the first version to be reviewed so it needs a full review. I've tried to re-use versions of dependencies previously reviewed in other projects.

CharlVS committed:
Repo: Fix project Pub.dev details

Make the description longer for Pub.dev and change the markdown checkboxes to emoji because they are rendered as a bullet list on Pub.dev.

CharlVS committed:
MVP: 0.0.1-preview.1

- [x] Cross-platform log storage
- [x] Cross-platform logs download
- [ ] Web multi-threading support
- [ ] Log levels (e.g. debug, info, warning, error)
- [ ] Performance metrics (in progress)
- [ ] Compressed file export
- [ ] Dev environment configurable logging filters for console
- [ ] Stacktrace formatting
- [ ] Log analytics

CharlVS committed:
0.1.1 - Bug fixes & Improvement

CharlVS committed:
Add hyperlink to README Komodo Logo

Navigate to KomodoPlatform GitHub profile when logo is clicked

CharlVS committed:
`1.0.0` - Stable release

CharlVS committed:
Add Pub.dev URL

CharlVS committed:
Add public API to clear all logs

CharlVS committed:
Refactor to share code and fix date parsing bug

Refactor to share more code between web and native platforms (focused mainly on file name and directory handling) and fix a bug where logs belonging to days with a single digit month or day could not be parsed.

CharlVS committed:
1.0.1 Release

1.0.1 Release

DeckerSU reviewed (approved):
In total LGTM. However, I would like to make a few notes:

- The File System Access API and the corresponding Dart library are experimental, so we should explicitly state this in the description - https://pub.dev/documentation/dragon_logs. We should also mention the supported browsers: Chrome, Edge, and Opera. Furthermore, since some methods like `remove` are marked as [experimental](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemHandle/remove) or non-standard, we should not rely on them. This should also be reflected in our documentation. 
- Regarding the `share_plus: ^7.0.2` package, if I'm not mistaken, we previously used a self-hosted version - https://github.com/KomodoPlatform/plus_plugins.git with the commit hash `052dcbb90aa2120c8f8384c05e17a82ad78a1758`. Perhaps we should use the same version here as well.

CharlVS commented:
Thanks for the review @DeckerSU.

1. I will add a note that some browsers have experimental support. The API has perhaps also matured in the 10 months since the packages release (which affects the package without an update needed). 
I see that the `.remove()` call is the only experimental call we are using for supported browsers. I will replace it with the `[removeEntry](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle/removeEntry#browser_compatibility)` API in the next release which has stable support.

2. For `share_plus`, I've had to reference the pub-hosted version because packages can't be published with git-hosted dependencies. Our internal security is the main priority so, alternatively, I can create a private fork with the git reference.
I'm also working on setting up a private Dart package hosting server for the org ([repo](https://github.com/KomodoPlatform/dart-bar/blob/main/README.md#why-host-your-own-pub-repository-with-dart-bar)).

takenagain committed:
Add padding to the month and day in the output file name

takenagain committed:
Add log export unit tests

takenagain committed:
Fix file order in file export for web and native

takenagain committed:
Export logs to file in test for visual confirmation

takenagain committed:
Clean log files after test with `tearDown` function

CharlVS committed:
Bump dependencies

NB: Although these changes resolve version conflicts for compatibility with the Flutter 2.22 release, they have not yet been refactored to work with the Wasm build of Flutter web.

CharlVS committed:
Merge branch 'fix/log-export-order' of https://github.com/KomodoPlatform/stored_logs

CharlVS committed:
v1.0.3: Fix logging order

takenagain committed:
Fix flutter analyze warnings

Using Flutter `3.16.9` to stick with the version used when initially creating the PR

takenagain committed:
Update dependencies to Flutter 3.22

CharlVS committed:
Merge branch 'fix/log-export-order' of https://github.com/KomodoPlatform/stored_logs

takenagain committed:
Fix log export ordering & upgrade Flutter (#2)

* Add padding to the month and day in the output file name

* Add log export unit tests

* Fix file order in file export for web and native

* Export logs to file in test for visual confirmation

* Clean log files after test with `tearDown` function

* Fix flutter analyze warnings

Using Flutter `3.16.9` to stick with the version used when initially creating the PR

* Update dependencies to Flutter 3.22

CharlVS committed:
Merge branch 'main' of https://github.com/KomodoPlatform/stored_logs

CharlVS committed:
Bump versions

CharlVS committed:
pubspec fix
